### ISSUE TITLE
Error when transforming field with `re` modifier

### ISSUE THREAD

**Issue by barvhaim** [SigmaHQ/pySigma#83]
Following the discussion on https://github.com/SigmaHQ/pySigma/issues/79, I managed to reproduce my issue

### Reproduce:
1. In `sigma.pipelines.elasticsearch.windows.ecs_windows` pipeline change `Image` mapping from `"Image": "process.executable",` to `"Image": ["process.executable"],` (line 106 for me)
2. Run the following (via cli or via code) - 
```
from sigma.rule import SigmaRule
from sigma.pipelines.elasticsearch.windows import ecs_windows


if __name__ == '__main__':
    rule = SigmaRule.from_yaml("""
        title: Test Rule
        status: experimental
        description: Test Rule
        references:
            - https://example.com
        logsource:
            category: process_creation
            product: windows
        detection:
            selection:
                Image|re: cmd.exe
            condition: selection
    """)
    pipeline = ecs_windows()
    rule = pipeline.apply(rule)
```

then you'll see the issue I am facing 
```
 raise SigmaTypeError(f"Modifier {self.__class__.__name__} incompatible to value type of '{ val }'", source=self.source)
sigma.exceptions.SigmaTypeError: Modifier SigmaRegularExpressionModifier incompatible to value type of 'SigmaRegularExpression(regexp='cmd.exe')'
```

when the mapping is 1:1 string it is working, but when there are multiple mapping it crashes for me. @thomaspatzke let me know if you manage to see it as well, my python is 3.9.13

**Comment by thomaspatzke** [comment]
Now I'm able to reproduce, thanks 😊

**Comment by thomaspatzke** [comment]
The `dataclasses.replace` you've mentioned in #79 causes that the `SigmaDetectionItem` is reinitialized with the already transformed values, which causes that they are transformed again. This can cause wrong values by multiply applying modifiers or in such restricted modifiers like `re` an exception.

**Comment by thomaspatzke** [comment]
Fixed! I'm preparing a release with some further minor fixes.

**Comment by thomaspatzke** [comment]
Fix will be contained in release 0.8.10-

**Comment by barvhaim** [comment]
Thanks! :)
